pymongo/asynchronous/client_session.py

```diff
@@ -222,6 +224,7 @@ def __init__(
                 )
         self._default_transaction_options = default_transaction_options
         self._snapshot = snapshot
+        self._bind = bind
```

Comment:
We have to bind/unbind the session in `ClientSession.__enter__/__exit__`. That way the stack of sessions is managed correctly (ie we call `_SESSION.reset(token)`). Think about how nested cases will work:
```python
session1 = client.start_session(bind=True)
with session1:
    session2 = client.start_session(bind=True)
    with session2:
        coll.find_one() # uses session2
    coll.find_one() # uses session1
coll.find_one() # uses implicit session
```

Reply:
Done in 3c68a70970e96917b0c418900203928bd7678a00